avm_scripts/docs-gen.sh

```diff
@@ -1,18 +1,21 @@
 #!/usr/bin/env bash
 
+function generate_docs() {
+  local dir=$1
+  echo "===> Generating examples documentation in " $dir
+  rm -f "$dir/.terraform.lock.hcl"
+  if [ -f "$dir/../../.terraform-docs.yml" ]; then
+    terraform-docs -c "$dir/../../.terraform-docs.yml" "$dir"
+  else
+    terraform-docs markdown table --output-file "$dir/README.md" --output-mode inject "$dir"
+  fi
+}
+
```

Comment:
I think this function should be called from the `examples/` directory. See below as well.

Also the `.terraform-docs.yml` file should be in examples/ not in the module root, as the config differs slightly.

```suggestion
function generate_docs() {
  local dir=$1
  echo "===> Generating examples documentation in " $dir
  rm -f "$dir/.terraform.lock.hcl"
  if [ -f ".terraform-docs.yml" ]; then
    terraform-docs -c ".terraform-docs.yml" "$dir"
  else
    terraform-docs markdown table --output-file "$dir/README.md" --output-mode inject "$dir"
  fi
}

```

Reply:
> I think this function should be called from the `examples/` directory. See below as well.
> 
> Also the `.terraform-docs.yml` file should be in examples/ not in the module root, as the config differs slightly.

We might have more config files like Checkov or Tflint configs, for those config files we don't have a better place to put them than the module root, so I think maybe it's better to keep all config files in the root. WDYT? @matt-FFFFFF

Reply:
The TF docs config for examples is different, so it lives in the examples directory. This is how we have it in the template.

We have 20+ repos all working this way so I think we should leave it.

Do you agree?